### PR TITLE
refactor(@ngtools/webpack): integrate compiler-cli Angular decorator downlevel transformer

### PR DESCRIPTION
**Pull request by clydin**
As of 10.0.0, the `@angular/compiler-cli` package provides a TypeScript transformer to downlevel Angular decorators.  This change switches the tooling webpack plugin from using an internal variant of the transformer to the new one within the compiler-cli.  For details on why this type of transformation is needed, please see: angular/angular@401ef71

### CODE REVIEW

File: packages/ngtools/webpack/src/angular_compiler_plugin.ts [hunk 1]
```diff
@@ -245,6 +245,11 @@ export class AngularCompilerPlugin {
         options.missingTranslation as 'error' | 'warning' | 'ignore';
     }
 
+    // For performance, disable AOT decorator downleveling transformer for applications in the CLI.
+    // The transformer is not needed for VE or Ivy in this plugin since Angular decorators are removed.
+    // While the transformer would make no changes, it would still need to walk each source file AST.
+    this._compilerOptions.annotationsAs = 'decorators' as 'decorators';
```
Comment (devversion) [inline packages/ngtools/webpack/src/angular_compiler_plugin.ts:251]: +💯!

**Comment by dgp1130** [comment]
@clydin, can you make a separate PR targetting patch? One of these commits is updating Angular to 10.0.0-rc.6, which probably shouldn't go to patch. I'm thinking this will be more complicated merge than just a cherry-pick.

**Comment by dgp1130** [comment]
@clydin, never mind I forgot path now refers to `10.0.x`. Merging it now.

**Comment by dgp1130** [comment]
Merged to `10.0.x` patch branch as well as `master`.

**Pull request merged**